### PR TITLE
Notes For Mar 16th

### PR DESCRIPTION
**Pull request by SriramRamesh**

### CODE REVIEW

File: mar16/sr5824.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+- Add random noise to the results from the data to protect individuals using 
+a probabilistic distribution
+- Privacy is violated by statistical database with any non-trivial utility 
+- Incase results cannot be randomized with a noise, we use an 
+exponential mechanism.Ex: Auctions. 
+
+
+# Open Questions and Future Work
+## Obladi
+- Scale independant of proxy
```
Comment (apanda) [inline mar16/sr5824.md]: Spelling: independant->independent

File: mar16/sr5824.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+- Leverage existing database logic for processing complex queries
```
Comment (apanda) [inline mar16/sr5824.md]: Can you expand on this, I am not sure I get it.

File: mar16/sr5824.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+- Limitations
+Benchmark has sub-optimal performance as it achieves GDPR compliance based on developer
+recommendation.
+
+## Differential Privacy
+- Add random noise to the results from the data to protect individuals using 
+a probabilistic distribution
+- Privacy is violated by statistical database with any non-trivial utility 
+- Incase results cannot be randomized with a noise, we use an 
+exponential mechanism.Ex: Auctions.
```
Comment (apanda) [inline mar16/sr5824.md]: Can you perhaps expand on this a bit.

File: mar16/sr5824.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+ Procedure for write and read are identical
+  1. check the path to access from position map
+  2. Read one obj from each bucket along that path
+  3. valid dummy object from that bucket using local
+     permutation map.
+  4. remap data to diff path (updated if write)
+  5. flushed later (i.e after reaching epoch)
+- Limitations:
+  1. Scalability and fault-tolerance when depending on a local centralized proxy
+  2. No Range or complex SQL queries
```
Comment (apanda) [inline mar16/sr5824.md:83]: One thing that might be interesting to add is whether you think applications would be willing to trade off some of this in exchange for ORAM's guarantees

**Review by apanda — COMMENTED** [review]
Some minor comments.

File: mar16/sr5824.md (hunk 2 above)
Comment (SriramRamesh) [reply mar16/sr5824.md]: Redis is a common key-value store. Obladi can leverage the implementation to write complex queries.

File: mar16/sr5824.md (hunk 3 above)
Comment (SriramRamesh) [reply mar16/sr5824.md]: McSherry and Talwar’s exponential mechanism outputs
price of an item at auction with a probability propotional to exp(−u(X,y)/2) where 
u(X,y) is the revenue if the actual price of item is y. This helps to enusre differential 
privacy.

File: mar16/sr5824.md (hunk 4 above)
Comment (SriramRamesh) [reply mar16/sr5824.md:83]: Obladi is a good choice if application require key value store that hides access pattern and provides ACID transactions with high latency.

**Review by apanda — APPROVED** [review]

**Pull request merged**